src/openrct2/paint/track/coaster/MultiDimensionRollerCoaster.cpp

```diff
@@ -5964,7 +6006,7 @@ static void MultiDimensionRCTrackLeftEighthToDiag(
                 PaintUtilSetSegmentSupportHeight(
                     session, PaintUtilRotateSegments(BlockedSegments::kStraightFlat, direction), 0xFFFF, 0);
                 MetalASupportsPaintSetup(
-                    session, kSupportType, MetalSupportPlace::Centre, 0, height + 36, session.SupportColours);
+                    session, supportType.metal, MetalSupportPlace::Centre, 0, height + 36, session.SupportColours);
```

Comment:
It seems a bit redundant to be passing `supportType.metal` to a function that's already called `MetalASupportsPaintSetup`. Is there a particular reason why this is not implied? I'm not too familiar with this part of the code, so it doesn't seem obvious to me.

Reply:
supportType is a union of WoodenSupportType and MetalSupportType, so we have to pass the correct one. I have not added an overload because there are quite a few functions for this kind of thing already, which would require 3-4 additional ones. I can change this if you really want, though.